websites/M/Musixmatch/presence.ts

```diff
@@ -1,45 +1,109 @@
 			startTimestamp: browsingTimestamp,
 		},
-		path = (text: string): boolean => {
-			return document.location.pathname.includes(text);
-		};
+		{ pathname, hostname } = document.location;
 
-	if (path("/explore/genre/")) {
-		presenceData.details = "Viewing a genre";
-		presenceData.state = document
-			.querySelector("#content > div > div.explore-panel > h1 > div")
-			.textContent.replace("lyrics", "")
-			.trim();
-	} else if (path("/explore")) presenceData.details = "Exploring lyrics";
-	else if (path("/lyrics/")) {
-		presenceData.details = document
-			.querySelector("#site h1")
-			.textContent.replace("Lyrics", "");
-		presenceData.state = document.querySelector(
-			"#site h2 > span > a"
-		).textContent;
-		presenceData.smallImageKey = "reading";
-	} else if (path("/community")) presenceData.details = "Viewing community";
-	else if (path("/profile/")) {
-		presenceData.details = "Viewing the profile:";
-		presenceData.state = document.querySelector("h1").textContent;
-	} else if (path("/artist/")) {
-		presenceData.details = "Viewing an artist:";
-		presenceData.state = document.querySelector("#content h1").textContent;
-	} else if (path("/search/")) {
-		presenceData.details = "Searching";
-		presenceData.state = document.querySelector(
-			"#content span > span"
-		).textContent;
-		presenceData.smallImageKey = "search";
+	if (hostname === "musixmatch.com" || hostname === "www.musixmatch.com") {
+		if (pathname === "/") presenceData.details = "Viewing the home page";
+		else if (pathname.includes("/explore"))
+			presenceData.details = "Exploring lyrics";
+		else if (pathname.includes("/community"))
+			presenceData.details = "Viewing the community page";
+		else if (pathname.includes("/contribute"))
+			presenceData.details = "Viewing the contribute page";
+		else if (pathname.includes("/profile/me")) {
+			//if user status = curator show curator badge
+			if (
+				document.querySelector(
+					"#site > div > div > div > main > div > div.mxm-user-header.mxm-user-header--profile > div > div:nth-child(1) > div:nth-child(2) > div > div > span"
+				)
+			) {
+				presenceData.smallImageKey = "https://i.imgur.com/Bl7BSIW.png";
+				presenceData.smallImageText = "Curator";
+			}
+			presenceData.details = "Viewing their profile";
+		} else if (pathname.includes("/profile/")) {
+			//if user status = curator show curator badge
+			if (
+				document.querySelector(
+					"#site > div > div > div > main > div > div.mxm-user-header.mxm-user-header--profile > div > div:nth-child(1) > div:nth-child(2) > div > div > span"
+				)
+			) {
+				presenceData.smallImageKey = "https://i.imgur.com/Bl7BSIW.png";
+				presenceData.smallImageText = "Curator";
+			}
+			presenceData.details = `Viewing ${
+				document.querySelector(
+					"#site > div > div > div > main > div > div.mxm-user-header.mxm-user-header--profile > div > div:nth-child(1) > div:nth-child(2) > div > div > h1"
+				).textContent
+			}'s profile`;
```

Comment:
Combine to not have duplicate code?

Reply:
This should be addressed

Reply:
Also idk, but If you can simplify the query selector to avoid breaking changes on the website...

Reply:
@RisingSunLight42 Yes, probably I can but I'm not at home, so I can change it in 3 days